### DISCUSSION TITLE
Help with efi

### DISCUSSION THREAD

**Discussion by qdog69** [ipxe/ipxe#716] [Q&A] [answered]
We have been using undionly.kpxe for about 5 years to boot Windows 10 on Dell workstations from virtual hard drives located on a Win 2012 Server and it has worked perfectly.

We got new workstations in and they are UEFI only and no longer support legacy boot. I was hoping it would be as easy as switching to snponly.efi but no such luck. I created the VHDX in windows and used diskpart to convert to GPT and add the appropriate efi, msr and system partitions. When it boots I am able to install Windows on the VHDX but after the install finishes and windows reboots I loose connection to the san.

We are using the latest ipxe as of 8/20/22 (1.21.1+)

Using Windows dhcp as follows:
	017 Root Path iscsi:1.1.1.1:::0:3090-Parent  (iPXE class)
	067 Bootfile Name ipxe.bat (iPXE class)
	067 Bootfile Name snponly.efi 

ipxe.bat contains:

#!ipxe

Set net0/keep-san 1

**As noted this config has always worked with legacy (using ipxe 1.0.0)

Debugging shows:

SAN 0x80.0 close: Connection reset (https://ipxe.org/0f0a6095) (works with legacy, no firewall)

Added in "netdevice" and also getting the following:
https://ipxe.org/err/3c3f63 (STP is disabled)
https://ipxe.org/err/2a0860

Anyone have an suggestions?

**Comment by mcb30** [comment]
It's not clear what error you are actually seeing, or even whether you're trying to describe an error message from iPXE or from Windows.

Could you please include a screenshot showing the error?

**Reply by stappersg** [reply]
As I read the support request, it summaries to _"but after the install finishes and windows reboots I loose connection to the san."_.
So something after _iPXE_ did do its job ...

**Reply by qdog69** [reply]
Unfortunately nothing comes off this network so no screen caps or pcaps. Time permitting, I can work on recreating it on a dev network

If I could post a screenshot it would say:
NETDEV net0 failed to receive 0x0: Operation not supported (https://ipxe.org/3f3c3f6383)

with the occasional 
NETDEV net0 unknown network protocol 9000
and
NETDEV net0 unknown network protocol aaaa

**Reply by mcb30** [reply]
Those are irrelevant errors corresponding to the receipt of unknown network packets. What is the (non-debug) error message that you see at the point of failure? Without this information, there is no way for anyone to help you.

**Reply by qdog69** [reply]
Without debugging there are no error messages, the Dell logo comes up like it is going to boot and never goes away

**Reply by NiKiZe** [reply]
Your current issue is after first stage windows install succeeding over iSCSI?

Have you tried doing a packet capture to see what happens on the wire? (I'm assuming that windows don't have any good logs to see connections working)

Also is your ipxe.bat the full file? What you have posted does in reality nothing.

**Reply by qdog69** [reply]
As best I can tell somethin is causing the iSCSI Virtual Disk Server service on Windows to crash (restart). I have done a capture and didn't notice anything serious and Windows logs are not much help, going though them now.

Yes that is my entire ipxe.bat file, as I mentioned this all works great for legacy...

**Reply by mcb30** [reply]
How are you starting the Windows installer?  It sounds as though you are relying on the (very very deprecated) `keep-san` to allow you to fall back to booting from a local installation medium such as a USB key.

Please consider `keep-san` to be totally unsupported.  I would be astonished if it worked at all under UEFI.  You may want to switch to using a network-based method for installing Windows: see e.g. https://ipxe.org/howto/winpe#installing_to_an_iscsi_target

**Reply by qdog69** [reply]
We install via CD but installs are rare, once we get the parent created we us diff disks.

**Reply by NiKiZe** [reply]
Please try to do an install on legacy without ipxe.bat to verify it working. Then try this on efi.

Reduce number of possible moving parts to make it easier to debug and reproduce.

**Comment by qdog69** [comment, marked as answer]
I spent the day rebuilding my iscsi server on Windows 2019 and the service is no longer crashing and I can boot to efi...

Thanks everyone for the responses.

**Reply by stappersg** [reply]
And as described in https://ipxe.org/howto/winpe#installing_to_an_iscsi_target or another method?

**Reply by qdog69** [reply]
No the original one I made worked fine on the new server so there was no need to try...

**Reply by stappersg** [reply]
So probably with `set net0/keep-san 1`, despite the warning:  _consider `keep-san` to be totally unsupported_

Please state expliciet that you, @qdog69, have a working combination of `keep-san` and _UEFI_.

**Reply by qdog69** [reply]
Yes I am using the ipxe.bat with "set net0/keep-san 1" and UEFI and it is working. 
Thanks for the reminder I will work on moving to the new technique.
